source-container-build/app/source_build.py

```diff
@@ -173,6 +173,16 @@ def fetch_image_manifest_digest(image: str) -> str:
     return run(cmd, check=True, text=True, capture_output=True).stdout.strip()
 
 
+# produces an artifact name that includes artifact's architecture
+# and repository id in the name
+def unique_srpm_artifact_name(file: str) -> str:
+    root, filename = os.path.split(file)
+    repoid = os.path.basename(root)
+    head, tail = os.path.split(root)
+    arch = os.path.basename(head)
+    return f"{arch}-{repoid}-{filename}"
```

Comment:
To avoid depending too much on the cachi2 output struture, IMO we can use something like `{checksum}-{filename}`. That should also better de-duplicate SRPMs that _should_ be de-duplicated